src/shogun/mathematics/graph/ops/shogun/Add.h

```diff
@@ -55,7 +56,35 @@ namespace shogun
 					if (CPU_arch->has_avx512f())
 						add_kernel_implementation_avx512f<T>(input1, input2, output, size);
 					else if (CPU_arch->has_avx())
-						add_kernel_implementation_avx<T>(input1, input2, output, size);
+					{
+						size_t i = 0;
+						if (size > 32/sizeof(T))
+						{
+							size_t remainder = size % (32/sizeof(T));
+							for (;i<size-remainder; i+=32/sizeof(T))
+							{
+								if constexpr(std::is_same_v<float, T>)
+								{
+									const auto packet1 = Eigen::internal::ploadu<Eigen::internal::Packet8f>(static_cast<const T*>(input1)+i);
+									const auto packet2 = Eigen::internal::ploadu<Eigen::internal::Packet8f>(static_cast<const T*>(input2)+i);
```

Comment:
`ploadu` would be `pload` if we can guarantee that memory is aligned. Right now we do 16 byte alignment which wouldn't work here, AVX requires 32-byte